internal/cli/release_list.go

```diff
@@ -0,0 +1,391 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/waypoint/internal/version"
+
+	"github.com/dustin/go-humanize"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/posener/complete"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	clientpkg "github.com/hashicorp/waypoint/internal/client"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+	serversort "github.com/hashicorp/waypoint/internal/server/sort"
+)
+
+type ReleaseListCommand struct {
+	*baseCommand
+
+	flagWorkspaceAll bool
+	flagVerbose      bool
+	flagUrl          bool
+	flagJson         bool
+	flagId           idFormat
+	filterFlags      filterFlags
+}
+
+func (c *ReleaseListCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithSingleApp(),
+	); err != nil {
+		return 1
+	}
+
+	// Get our API client
+	client := c.project.Client()
+
+	err := c.DoApp(c.Ctx, func(ctx context.Context, app *clientpkg.App) error {
+		var wsRef *pb.Ref_Workspace
+		if !c.flagWorkspaceAll {
+			wsRef = c.project.WorkspaceRef()
+		}
+
+		// Get the latest release
+		release, err := client.GetLatestRelease(ctx, &pb.GetLatestReleaseRequest{
+			Application: app.Ref(),
+			Workspace:   c.project.WorkspaceRef(),
+		})
+		if status.Code(err) == codes.NotFound {
+			err = nil
+			release = nil
+		}
+		if err != nil {
+			app.UI.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return ErrSentinel
+		}
```

Comment:
I think we can delete this `GetLatestRelease` API call all together. The `UI_ListReleases` gives us the latest release. I don't remember what order `UI_ListReleases` does by default but there's an Order field too! Then we can grab the latest release off of the `Releases` slice to be used later on line 138 (not sure if there's another place this is used to check against the latest release but that's one of them).